I first tried this out from distribution/zip, maybe a missing dependency?

```
:distribution:zip:integTest#clean UP-TO-DATE
:distribution:zip:integTest#extract

FAILURE: Build failed with an exception.

* What went wrong:
Cannot expand ZIP '/Users/rmuir/workspace/elasticsearch/distribution/zip/build/distributions/elasticsearch-3.0.0-SNAPSHOT.zip' as it does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

I'm trying from the top-level build now.

--- a/build.gradle
+++ b/build.gradle
@@ -142,6 +142,7 @@ subprojects {
             substitute module("org.elasticsearch:test-framework:${version}") with project("${projectsPrefix}:test-framework")
           }
           substitute module("org.elasticsearch.distribution.zip:elasticsearch:${version}") with project("${projectsPrefix}:distribution:zip")
+          substitute module("org.elasticsearch.distribution.tar:elasticsearch:${version}") with project("${projectsPrefix}:distribution:tar")
         }
       }
     }

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterConfiguration.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterConfiguration.groovy
@@ -36,6 +36,9 @@ class ClusterConfiguration {
     @Input
     int transportPort = 9500
 
+    @Input
+    String jvmArgs = System.getProperty('tests.jvm.argline', '')
+
     Map<String, String> systemProperties = new HashMap<>()
 
     @Input

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
@@ -61,6 +61,24 @@ class ClusterFormationTasks {
         File pidFile = pidFile(baseDir)
         String clusterName = "${task.path.replace(':', '_').substring(1)}"
         File home = homeDir(baseDir, config.distribution)
+        Map esConfig = [
+            'cluster.name': clusterName,
+            'http.port': config.httpPort,
+            'transport.tcp.port': config.transportPort,
+            'pidfile': pidFile,
+            // TODO: make this work for multi node!
+            'discovery.zen.ping.unicast.hosts': "localhost:${config.transportPort}",
+            'path.repo': "${home}/repo",
+            'path.shared_data': "${home}/../",
+            // Define a node attribute so we can test that it exists
+            'node.testattr': 'test',
+            'repositories.url.allowed_urls': 'http://snapshot.test*'
+        ]
+        Map esEnv = [
+            'JAVA_HOME': System.getProperty('java.home'),
+            'ES_GC_OPTS': config.jvmArgs
+        ]
+
         List setupDeps = [] // need to copy the deps, since start will later be added, which would create a circular task dep!
         setupDeps.addAll(task.dependsOn)
         Task setup = project.tasks.create(name: "${task.name}#clean", type: Delete, dependsOn: setupDeps) {
@@ -71,19 +89,6 @@ class ClusterFormationTasks {
         setup = project.tasks.create(name: "${task.name}#configure", type: DefaultTask, dependsOn: setup) << {
             File configFile = new File(home, 'config/elasticsearch.yml')
             logger.info("Configuring ${configFile}")
-            Map esConfig = [
-                'cluster.name': clusterName,
-                'http.port': config.httpPort,
-                'transport.tcp.port': config.transportPort,
-                'pidfile': pidFile,
-                // TODO: make this work for multi node!
-                'discovery.zen.ping.unicast.hosts': "localhost:${config.transportPort}",
-                'path.repo': "${home}/repo",
-                'path.shared_data': "${home}/../",
-                // Define a node attribute so we can test that it exists
-                'node.testattr': 'test',
-                'repositories.url.allowed_urls': 'http://snapshot.test*'
-            ]
             configFile.setText(esConfig.collect { key, value -> "${key}: ${value}" }.join('\n'), 'UTF-8')
         }
         for (Map.Entry<String, String> command : config.setupCommands.entrySet()) {
@@ -129,12 +134,13 @@ class ClusterFormationTasks {
                 throw new GradleException('Failed to start elasticsearch')
             }
         }
-        Task start;
+        Task start
         if (Os.isFamily(Os.FAMILY_WINDOWS)) {
             // elasticsearch.bat is spawned as it has no daemon mode
             start = project.tasks.create(name: "${task.name}#start", type: DefaultTask, dependsOn: setup) << {
                 // Fall back to Ant exec task as Gradle Exec task does not support spawning yet
                 ant.exec(executable: 'cmd', spawn: true, dir: home) {
+                    esEnv.each { env(key: key, value: value) }
                     (['/C', 'call', 'bin/elasticsearch'] + esArgs).each { arg(value: it) }
                 }
                 esPostStartActions(ant, logger)
@@ -145,6 +151,7 @@ class ClusterFormationTasks {
                 executable 'sh'
                 args 'bin/elasticsearch', '-d' // daemonize!
                 args esArgs
+                environment esEnv
                 errorOutput = new ByteArrayOutputStream()
                 doLast {
                     if (errorOutput.toString().isEmpty() == false) {

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
@@ -228,11 +228,12 @@ class ClusterFormationTasks {
 
     static void configureDistributionDependency(Project project, String distro) {
         String elasticsearchVersion = ElasticsearchProperties.version
+        String packaging = distro == 'tar' ? 'tgz' : distro
         project.configurations {
             elasticsearchDistro
         }
         project.dependencies {
-            elasticsearchDistro "org.elasticsearch.distribution.${distro}:elasticsearch:${elasticsearchVersion}@${distro}"
+            elasticsearchDistro "org.elasticsearch.distribution.${distro}:elasticsearch:${elasticsearchVersion}@${packaging}"
         }
     }
 }

--- a/distribution/build.gradle
+++ b/distribution/build.gradle
@@ -41,8 +41,10 @@ allprojects {
 }
 
 subprojects {
+  /*****************************************************************************
+   *                            Rest test config                               *
+   *****************************************************************************/
   apply plugin: 'elasticsearch.rest-test'
-
   integTest {
     includePackaged true
   }

--- a/distribution/deb/build.gradle
+++ b/distribution/deb/build.gradle
@@ -20,6 +20,9 @@
 task buildDeb(type: Deb) {
   dependsOn dependencyFiles
 }
+
 artifacts {
   archives buildDeb
 }
+
+integTest.dependsOn buildDeb

--- a/distribution/rpm/build.gradle
+++ b/distribution/rpm/build.gradle
@@ -20,6 +20,9 @@
 task buildRpm(type: Rpm) {
   dependsOn dependencyFiles
 }
+
 artifacts {
   archives buildRpm
 }
+
+integTest.dependsOn buildRpm

--- a/distribution/tar/build.gradle
+++ b/distribution/tar/build.gradle
@@ -24,5 +24,11 @@ task buildTar(type: Tar, dependsOn: dependencyFiles) {
 }
 
 artifacts {
+  'default' buildTar
   archives buildTar
+}
+
+integTest {
+  dependsOn buildTar
+  clusterConfig.distribution = 'tar'
 }

--- a/distribution/zip/build.gradle
+++ b/distribution/zip/build.gradle
@@ -27,3 +27,5 @@ artifacts {
   archives buildZip
 }
 
+integTest.dependsOn buildZip
+

